### DISCUSSION TITLE
Having an idle state vs lazily starting the machine

### DISCUSSION THREAD

**Discussion by lemuelflores** [statelyai/xstate#5007] [Q&A]
The simplest example but not limited to that I could probably think of is a plain fetch that needs a certain variable.

Say the use case we want is for an actor that fetches a specific resource.

Would you have an idle state that listens for a "GET_BY_ID" kind of event?

Or would just lazily create the machine and passing "input"?

**Comment by davidkpiano** [comment]
It depends on what you want to do, but "waiting for a value" is still a state IMO. Having an idle/pending state makes sense to me.
